framework/include/auxkernels/NodalPatchRecovery.h

```diff
@@ -0,0 +1,130 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef NODALPATCHRECOVERY_H
+#define NODALPATCHRECOVERY_H
+
+#include "AuxKernel.h"
+#include "FEProblem.h"
+
+/**
+ * This class uses patch recovery technique to recover material properties to smooth nodal fields
+ * Zienkiewicz, O. C. and Zhu, J. Z. (1992), The superconvergent patch recovery and a posteriori
+ * error estimates. Part 1: The recovery technique. Int. J. Numer. Meth. Engng., 33: 1331-1364.
+ * doi:10.1002/nme.1620330702
+ */
```

Comment:
Add this to the description:
If used with elemental (monomial) AuxVariables, this class falls back on the standard techniques in AuxKernel.C to compute either constant or higher-order monomials, depending on the order of the AuxVariable.